pkg/parse/events/events.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"context"
+)
+
+// ScheduleFunc generates a list of EventProducer, each with their own event
+// generating channel and event handler function.
+type ScheduleFunc func(context.Context) []EventProducer
+
+// HandleEventFunc handles an Event and decides whether to call the RunFunc.
+type HandleEventFunc func(Event) HandleResult
+
+// HandleResult encapsulates the result of a HandleEventFunc.
+// This simply allows explicitly naming return values in a way that makes the
+// implementation easier to read.
+type HandleResult struct {
+	// DelayStatusUpdate tells GenerateFunc to reset the status update timer.
+	DelayStatusUpdate bool
+	// ResetRetryBackoff tells GenerateFunc to reset the retry backoff.
+	ResetRetryBackoff bool
+}
+
+// Event represents the cause of HandleEventFunc being called.
+// Some event types may have special fields or methods.
+type Event interface {
```

Comment:
None of these events have any special fields any more, now that I added the ResultHandler to the EventProducer to allow all producers to inject generic even cleanup behavior.

So the events could probably be replaced with strings, as long as we don't need to add any other event-specific inputs.